pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/upsertcompaction/UpsertCompactionTaskGenerator.java

```diff
@@ -160,6 +170,20 @@ public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
     return pinotTaskConfigs;
   }
 
+  private String getDefaultValidDocId(UpsertConfig upsertConfig) {
```

Comment:
@Jackie-Jiang I found that the snapshot is only available if enableSnapshot=true. So, we need to check the upsert config and pick the validDocIdType appropriately.

Please check this logic.

Reply:
updated

1. throw error if enableSnapshot is not set to true.
2. throw error if queryableDocId type is set while upsert delete column is not enabled.